pkgs/stdenv/generic/setup.sh

```diff
@@ -834,9 +834,24 @@ substituteStream() {
     local description=$2
     shift 2
 
+    local has_warned_replace_deprecation=
     while (( "$#" )); do
+        local is_required=1
+        local is_quiet=
         case "$1" in
+            --replace-quiet)
+                is_quiet=1
+                ;&
             --replace)
+                if [ -z "$has_warned_replace_deprecation" ]; then
+                    echo "substituteStream(): WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (in $description)" >&2
+                    has_warned_replace_deprecation=1
+                fi
+                ;&
+            --replace-warn)
+                is_required=
```

Comment:
Is this an "unset"?

Reply:
No, its setting it to empty, to trigger the `[ -z x ]` test.
Would you prefer this?
```suggestion
                is_required=""
```

Reply:
Yes. Shell script is known to be a very sensitive language.